pkg/build/webhook/gitlab/gitlab_test.go

```diff
@@ -0,0 +1,363 @@
+package gitlab
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/build/webhook"
+)
+
+var testBuildConfig = &api.BuildConfig{
+	Spec: api.BuildConfigSpec{
+		Triggers: []api.BuildTriggerPolicy{
+			{
+				Type: api.GitLabWebHookBuildTriggerType,
+				GitLabWebHook: &api.WebHookTrigger{
+					Secret: "secret101",
+				},
+			},
+			{
+				Type: api.GitLabWebHookBuildTriggerType,
+				GitLabWebHook: &api.WebHookTrigger{
+					Secret: "secret100",
+				},
+			},
+			{
+				Type: api.GitLabWebHookBuildTriggerType,
+				GitLabWebHook: &api.WebHookTrigger{
+					Secret: "secret102",
+				},
+			},
+		},
+		CommonSpec: api.CommonSpec{
+			Source: api.BuildSource{
+				Git: &api.GitBuildSource{
+					URI: "git://github.com/my/repo.git",
```

Comment:
I think this can be any URI? I think the intended use is a custom gitlab server.